client/components/data/query-rewind-backup-status/index.js

```diff
@@ -1,49 +1,35 @@
+/** @format */
 /**
  * External dependencies
- *
- * @format
  */
 
-import { PureComponent } from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { delay } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { getRewindBackupProgress } from 'state/activity-log/actions';
+import Interval, { EVERY_SECOND } from 'lib/interval';
+import { getRewindBackupProgress as getBackupProgress } from 'state/activity-log/actions';
 
-class QueryRewindBackupStatus extends PureComponent {
+class QueryRewindBackupStatus extends Component {
 	static propTypes = {
-		freshness: PropTypes.number,
 		downloadId: PropTypes.number,
 		siteId: PropTypes.number.isRequired,
 	};
 
-	query( props ) {
-		const { getBackupProgress, downloadId, siteId } = props;
-		if ( siteId && downloadId ) {
-			delay( getBackupProgress, 1500, siteId, downloadId );
-		}
-	}
-
-	componentWillMount() {
-		this.query( this.props );
-	}
+	query = () => {
+		const { downloadId, siteId } = this.props;
 
-	componentWillUpdate( nextProps ) {
-		const { freshness, downloadId } = this.props;
-		if ( downloadId !== nextProps.downloadId || freshness !== nextProps.freshness ) {
-			this.query( nextProps );
+		if ( siteId && downloadId ) {
+			this.props.getBackupProgress( siteId, downloadId );
 		}
-	}
+	};
 
 	render() {
-		return null;
+		return <Interval onTick={ this.query } period={ EVERY_SECOND } />;
```

Comment:
@samouri @gwwar @coderkevin @nosolosw cc'ing y'all because this is one way I'm achieving polling via a query component - hopefully in the short-term until we have a reliable `freshness` built-in

thoughts?

Reply:
makes sense to me. ideally qc's would have accept `period` as a configurable prop while having a reasonable default (1 day)

Reply:
Sounds ok to me. every second may be a bit aggressive for folks with slow internet though.

Reply:
> ideally qc's would have accept period as a configurable prop while having a reasonable default (1 day)

I'll leave that up to you as you auto-generate them @samouri :wink:

> every second may be a bit aggressive

so it's awkward because this updated every second but the data layer is throttling at 1.5s. that would be better if freshness were a thing. I wanted to do 3s but these processes move so quickly that I felt we might miss most of the bar and it would go from like 5% to 100% `¯\_(ツ)_/¯`

Reply:
Got it. Just keep an 👀  out while testing. 

Another option might be to use more of a binary state for displaying UI (blinking dot/done) instead of a progress bar.